ni_measurementlink_service/_drivers/__init__.py

```diff
@@ -4,25 +4,47 @@
 import contextlib
 from typing import ContextManager, TypeVar
 
+from ni_measurementlink_service.session_management._types import SessionInitializationBehavior
+
 TSession = TypeVar("TSession")
 
+_EXISTING_BEHAVIORS = [
+    SessionInitializationBehavior.AUTO,
+    SessionInitializationBehavior.ATTACH_TO_SERVER_SESSION,
+    SessionInitializationBehavior.INITIALIZE_SERVER_SESSION,
+]
+
 
-def closing_session(session: TSession) -> ContextManager[TSession]:
+def closing_session(
+    initialization_behavior: SessionInitializationBehavior,
+    session: TSession,
+) -> ContextManager[TSession]:
     """Create a context manager that closes the session.
 
     Args:
+        initialization_behavior: Specifies whether to initialize a new
+            session or attach to an existing session.
+
         session: A driver session.
 
     Returns:
         A context manager that yields the session and closes it.
+
+    Raises:
+        TypeError: If the session is not a context manager and if it does not have a close() method.
+
+        ValueError: If the initialization behavior is invalid.
```

Comment:
This is an internal function, so its docstring doesn't need all this detail.

What it does need is to say that this function emulates the behavior of INITIALIZE_SESSION_THEN_DETACH and ATTACH_SESSION_THEN_CLOSE.